compio-net/src/socket.rs

```diff
@@ -319,6 +323,45 @@ impl Socket {
         compio_runtime::submit(op).await.into_inner()
     }
 
+    #[cfg(unix)]
+    pub fn get_socket_option<T>(&self, level: i32, name: i32) -> io::Result<T> {
```

Comment:
This is unsafe without any constraint on T

Reply:
I'll mark them (this and `set_socket_option`) as unsafe. It's not easy to put safety bound on T. Popular crates like `nix` and `rustix` provide safe interfaces but they are too complex and not necessary for us to implement again.

Reply:
Yeah I think an unsafe mark and some safety note should be enough for us.